spec/services/discovery_engine/evaluation/evaluation_resource_spec.rb

```diff
@@ -0,0 +1,60 @@
+RSpec.describe DiscoveryEngine::Evaluation::EvaluationResource do
+  let(:sample_set_id) { "clickstream_01_02" }
+  let(:evaluation_resource) { described_class.new(sample_set_id) }
+
+  let(:evaluation_service_stub) { double("evaluation_service", create_evaluation: operation_object, get_evaluation: evaluation_object) }
+  let(:operation_object) { double("operation", error?: false, wait_until_done!: true, results: response_object) }
+  let(:response_object) { double("response", name: "/evaluations/1") }
+  let(:evaluation_object) { double("evaluation", state: :SUCCEEDED, quality_metrics: quality_metrics_object) }
+  let(:quality_metrics_object) { double("quality_metrics", to_h: "some output") }
+
+  before do
+    allow(DiscoveryEngine::Clients).to receive(:evaluation_service).and_return(evaluation_service_stub)
+    allow(Rails.logger).to receive(:info)
+  end
+
+  describe "#fetch_quality_metrics" do
+    it "calls the create_evaluation endpoint" do
+      evaluation_resource.fetch_quality_metrics
+
+      expect(evaluation_service_stub).to have_received(:create_evaluation).with(
+        parent: Rails.application.config.discovery_engine_default_location_name,
+        evaluation: {
+          evaluation_spec: {
+            query_set_spec: {
+              sample_query_set: "#{Rails.application.config.discovery_engine_default_location_name}/sampleQuerySets/#{sample_set_id}",
+            },
+            search_request: {
+              serving_config: ServingConfig.default.name,
+            },
+          },
+        },
+      )
+
+      expect(Rails.logger).to have_received(:info)
+        .with("Successfully created evaluation /evaluations/1")
+    end
+
+    it "polls the get_evaluation endpoint until state == :SUCCEEDED" do
```

Comment:
Might be worth having the stub return `:PENDING` the first time to test the loop?

Reply:
Ah maybe this https://stackoverflow.com/questions/1168845/rspec-stubbing-kernelsleep I'll give that a go. Hold the line caller!

Reply:
> Ah that doesn't help, it's that the test becomes stuck in the while loop so you need to change its state from :PENDING to :SUCCEEDED.

Yeah, something like this maybe?
```ruby
allow(evaluation_service_stub).to receive(:get_evaluation)
  .and_return(pending_response)
  .and_return(pending_response)
  .and_return(successful_response)
```